src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -127,17 +92,17 @@ class AutomaticSpeechRecognitionPipeline(ChunkPipeline):
     to support multiple audio formats
     """
 
-    def __init__(self, feature_extractor: Union["SequenceFeatureExtractor", str], *args, **kwargs):
+    def __init__(self, model, tokenizer, feature_extractor: Union["SequenceFeatureExtractor", str], *args, **kwargs):
```

Comment:
Is that a bit backwards breaking? Think it's totally fine for me as most people probably just use the pipeline function anyways.

Reply:
It is breaking in some sense yes.

Calls like `AutomaticSpeechRecognitionPipeline(feature_extractor, model, tokenizer)` will break.
But, the regular pipeline has a call signature `(model, tokenizer, feature_extractor)`, so this makes it a little bit consistent.
It also makes the parent class responsible of doing `self.feature_extractor = feature_extractor` (so less risk of future discrepancy between pipelines). It's definitely something that should probably be a separate PR, and make sure ALL PR henceforth have a correct call signature. We could also use something like `MyPipeline(*, model. tokenizer, feature_extractor)` which would disallow purely position based arguments, which would make everything simpler to maintain IMO with reducing confusion for users too: https://www.vegardstikbakke.com/python-keyword-only/

I will revisit the PR to exclude this from the current PR, it's definitely not good just putting that here, I would rather make a sweep at all classes and make a solid (tested) decision about parameter flow. (I think the current thing, is just something that was written before the rewrite of pipelines and just happened to be that way, not a conscious decision.